portal-api/src/v1/gef/models/facilities.js

```diff
@@ -63,6 +74,12 @@ class Facility {
       }
       // used to store the user details of maker who changed unissued facility to issued
       this.unissuedToIssuedByMaker = Object(req.unissuedToIssuedByMaker) || null;
+      if (isFacilityEndDateEnabledOnGefVersion(dealVersion)) {
+        if ('isUsingFacilityEndDate' in req && !(typeof req.isUsingFacilityEndDate === 'boolean')) {
+          throw new InvalidParameterError('isUsingFacilityEndDate', req.isUsingFacilityEndDate);
+        }
+        this.isUsingFacilityEndDate = req.isUsingFacilityEndDate ?? null;
```

Comment:
Should we test for if 
facility end date does not exist -> throw
if its not boolean -> throw

then we know this is always correctly set